### PR TITLE
Feat/user config refactor

### PR DESCRIPTION
**Pull request by kaliberpoziomka**
UserConfiguration library refactor, so that it operates on structs and not reading/writing to/from PoolStorage.

### CODE REVIEW

File: contracts/protocol/libraries/configuration/user_configuration.cairo [hunk 1]
```diff
@@ -223,7 +223,7 @@ namespace UserConfiguration:
     # @return The address of the only asset used as collateral
     # @return The debt ceiling of the reserve
     func get_isolation_mode_sate{syscall_ptr : felt*, pedersen_ptr : HashBuiltin*, range_check_ptr}(
```
Comment (msaug) [inline contracts/protocol/libraries/configuration/user_configuration.cairo:225]: typo `get_isolation_mode_sate`

**Comment by msaug** [comment]
I think we should not merge this because when you call `set_x`, it actually doesn’t set anything, you’re just returned an updated value that you need to store somewhere afterward.

In solidity, the functions take UserConfigurationMap as a parameter, but as a storage variable. So it’s directly written in storage when updated, which is not the case in Cairo. If we want to replicate the behavior, we should keep it as it is right now, writing in PoolStorage inside the UserConfiguration library.

**Comment by EdgarBarrantes** [comment]
The alternative to this would be to create a `user_configuration_library.cairo` with something like `get_updated_x` that would return the values and set them in the main file. But perhaps that would be too much?

Also, test didn't pass due to formatting, you can install the pre-commit hook (`./pre-commit-install.sh`) and/or if you're using vs code use the [cairo extension](https://marketplace.visualstudio.com/items?itemName=starkware.cairo) and set format on save.

**Comment by kaliberpoziomka** [comment]
After some thought and discussion, I delete this PR and I'll create a new one, to just finish testing. The UserConfiguration will write structs to PoolStorage, which is more consistent with Solidity implementation and also more logical according to how the library is used and how functions are called.

**Comment by achab** [comment]
Thanks for this PR @kaliberpoziomka, it's not clear which behaviour we should follow. I also think this "library" contract should not write to storage. IMO they should be just be util functions called by "contract logic code", and the latter should write to storage. But it's indeed safer to follow solidity approach.

**Pull request closed without merging**